cooperative_lanechange/src/cooperative_lanechange.cpp

```diff
@@ -310,83 +347,109 @@ namespace cooperative_lanechange
         header.recipient_id = DEFAULT_STRING_;  
         header.sender_bsm_id = bsmIDtoString(bsm_core_);
         header.plan_id = boost::uuids::to_string(boost::uuids::random_generator()());
-        header.timestamp = trajectory_plan.front().target_time.toNSec();
+        header.timestamp = trajectory_plan.front().target_time.toNSec() *1000000;
         request_msg.header = header;
 
         request_msg.strategy = "carma/cooperative-lane-change";
         request_msg.plan_type.type = cav_msgs::PlanType::CHANGE_LANE_LEFT;
         //Urgency- Currently unassigned
 
-        //Location
-        cav_msgs::LocationECEF location;
-        location.ecef_x = int(pose_msg_.pose.position.x);
-        location.ecef_y = int(pose_msg_.pose.position.y);
-        location.ecef_z = int(pose_msg_.pose.position.z);
-        //Using trajectory first point time as location timestamp
-        location.timestamp = trajectory_plan.front().target_time.toNSec();
-        request_msg.location = location;
         //Strategy params
         //Encode JSON with Boost Property Tree
         using boost::property_tree::ptree;
         ptree pt;
-        pt.put("speed",maneuver.lane_change_maneuver.end_speed); 
-        pt.put("start_lanelet",maneuver.lane_change_maneuver.starting_lane_id);
-        pt.put("end_lanelet", maneuver.lane_change_maneuver.ending_lane_id);
+        double end_speed_floor = std::floor(maneuver.lane_change_maneuver.end_speed);
+        int end_speed_fractional = (maneuver.lane_change_maneuver.end_speed - end_speed_floor) * 10;
+
+        ROS_DEBUG_STREAM("end_speed_floor" << end_speed_floor);
+        ROS_DEBUG_STREAM("end_speed_fractional" << end_speed_fractional);
+        ROS_DEBUG_STREAM("start_lanelet_id" << maneuver.lane_change_maneuver.starting_lane_id);
+        ROS_DEBUG_STREAM("end_lanelet_id" << maneuver.lane_change_maneuver.ending_lane_id);
+
+        pt.put("s",(int)end_speed_floor); 
+        pt.put("f",end_speed_fractional); 
```

Comment:
Did we shorten these just to save space in the message? Looks a lot harder to interpret what these are now, it would be nice to revert them to meaningful names if possible.

Reply:
Or maybe - if not - thats an indicator that we need to increase the max payload size for our mobility messages again.

Reply:
The latter, the messages dont encode with the proper names in the string.
@MishkaMN

Reply:
Good to know, I'll write up a story for that then and we can consider it for later.